tracing/tracing.go

```diff
@@ -0,0 +1,55 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/HomesNZ/env"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/exporters/trace/jaeger"
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+// TracerConfig manages initialization of the tracing collector
+type TracerConfig struct {
+	Name          string
+	CollectorHost string
+	CollectorPort string
+	CollectorPath string
+}
+
+func (cfg TracerConfig) collectorEndpoint() string {
+	return fmt.Sprintf("http://%s:%s%s", cfg.CollectorHost, cfg.CollectorPort, cfg.CollectorPath)
+}
+
+// TracerConfigFromEnv initializes a TracerConfig from environment variables.
+func TracerConfigFromEnv() TracerConfig {
+	return TracerConfig{
+		Name:          env.MustGetString("SERVICE_NAME"),
+		CollectorHost: env.MustGetString("TRACING_COLLECTOR_HOST"),
+		CollectorPort: env.MustGetString("TRACING_COLLECTOR_PORT"),
+		CollectorPath: env.MustGetString("TRACING_COLLECTOR_PATH"),
+	}
+}
+
+// InitTracer adds a standard Jaeger Tracer to the otel global API
+func InitTracer(ctx context.Context, cfg TracerConfig, sampleType trace.Sampler) (func(), error) {
+	flush, err := jaeger.InstallNewPipeline(
+		jaeger.WithCollectorEndpoint(cfg.collectorEndpoint()),
```

Comment:
if you use URL, you should be able to call https://golang.org/pkg/net/url/#URL.RequestURI method.

Reply:
Alternatively, you can set it all up in environment variable JAEGER_ENDPOINT, and it will be initialised https://github.com/open-telemetry/opentelemetry-go/blob/63a11144cf5a5fd185e40eba7dd22bce2b2980ef/exporters/trace/jaeger/env.go#L45